course05/homework02/freekpaans/linked_list.cc

```diff
@@ -0,0 +1,74 @@
+#include <string>
+#include <numeric>
+
+#include "linked_list.h"
+
+static void push_at(LinkedList::LinkPtr& link, const std::string& str)
+{
+	LinkedList::LinkPtr newLink{new LinkedList::Link { str, {} }};
+
+	newLink->mNext = std::move(link);
```

Comment:
I guess  I could move the `std::move` into the `new` above, but I wasn't 100% sure if that would be exception safe. I _think_ it would, as the `std::move` only this does something when it actually assigns, right, and for that it needs to have the memory allocated.

Reply:
I think this is safe too